core/src/main/java/com/google/instrumentation/stats/MeasurementDescriptor.java

```diff
@@ -22,47 +24,34 @@
  *
  * <p>Note: MeasurementDescriptor names are {@link String}s with enforced restrictions.
  */
-public final class MeasurementDescriptor {
+@AutoValue
+public abstract class MeasurementDescriptor {
   public static final int MAX_LENGTH = StringUtil.MAX_LENGTH;
 
+  MeasurementDescriptor() {}
```

Comment:
I thought that that type of comment would become redundant if we did the same thing for all value classes.  Do you think it is still worth documenting within each class?